test/cpp/jit/test_gpu.cpp

```diff
@@ -6610,6 +6610,328 @@ TEST(NVFuserTest, FusionSmemDynamicPersistentSoftmax2D_CUDA) {
+                                          x_mean_sub_pow,
+                                          var_sum_bcast,
+                                          var,
+                                          var_eps,
+                                          rvar,
+                                          output});
+
+  std::vector<TensorView*> all_tensors({input});
+  all_tensors.insert(
+      all_tensors.end(), reduction_tensors.begin(), reduction_tensors.end());
+  all_tensors.insert(
+      all_tensors.end(), other_tensors.begin(), other_tensors.end());
+
+  const size_t outermost_non_reduction_dim = input->nDims() - norm_shape.size();
+  for (auto tv : all_tensors) {
+    // Flatten normalization shape into inner-most dimension
+    for (size_t idx = 0; idx < norm_shape.size() - 1; ++idx) {
+      tv->merge(-2, -1);
+    }
+    // Flatten remaining dimensions into outer-most dimension
+    for (size_t idx = 0; idx < outermost_non_reduction_dim - 1; ++idx) {
+      tv->merge(0, 1);
+    }
+  }
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn(input_shape, options);
+
+  // Check reduction axis is same for all reductions
+  // Generate Launch Parameters
+  auto reduction_params =
+      getMultipleReductionHeuristics(&fusion, {t0}, reduction_tensors);
+  TORCH_CHECK(reduction_params, "Reduction schedule was not generated!");
+
+  scheduleMultipleReduction(
+      &fusion, reduction_params.value(), reduction_tensors, other_tensors);
+
+  torch::jit::fuser::cuda::FusionExecutor fe;
+  fe.compileFusion(&fusion);
+  auto outputs = fe.runFusion({t0}, reduction_params.value().lparams);
+
+  auto result = at::layer_norm(t0, norm_shape);
+  TORCH_CHECK(
+      result.allclose(outputs[0], 1e-4, 1e-4),
+      "Error of: ",
+      result.sub(outputs[0]).abs().max());
+}
+
+TEST(NVFuserTest, FusionMagicSchedulerBatchNormalization_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  const float kMomentum = 0.1;
+  const float kEps = 1e-5;
+  c10::IntArrayRef input_shape{20, 100, 35, 45};
+
+  auto input = makeSymbolicTensor(input_shape.size());
+  auto weight = makeSymbolicTensor(1);
+  auto bias = makeSymbolicTensor(1);
+  fusion.addInput(input);
+  fusion.addInput(weight);
+  fusion.addInput(bias);
+  // auto running_mean = makeSymbolicTensor(1);
+  // auto running_var = makeSymbolicTensor(1);
+  // fusion.addInput(running_mean);
+  // fusion.addInput(running_var);
+
+  const int kNumberOfDims = input->nDims();
+  std::vector<int> reduction_axes;
+  std::vector<bool> broadcast_mask(kNumberOfDims, false);
+  Val* num_features = nullptr;
+
+  for (size_t axis = 0; axis < kNumberOfDims; ++axis) {
+    if (axis != 1) {
+      reduction_axes.push_back(axis);
+      broadcast_mask[axis] = true;
+      num_features = (axis == 0)
+          ? input->domain()->domain()[0]->extent()
+          : mul(num_features, input->domain()->domain()[axis]->extent());
```

Comment:
For layer norm, we are given a list of ints, which represents the number of features for normalization. I can calculate that in parser.cpp. On second thought, this would require generating a new kernel for a new size, so this dynamic approach is better. :man_shrugging: